optimum/neuron/utils/hub_neuronx_cache.py

```diff
@@ -364,25 +417,153 @@ def get_hub_cached_entries(
     api = HfApi(endpoint=endpoint, token=token)
     repo_files = api.list_repo_files(cache_repo_id)
     # Get the config corresponding to the model
-    target_entry = ModelCacheEntry(model_id, (AutoConfig.from_pretrained(model_id)))
+    try:
+        config = AutoConfig.from_pretrained(model_id)
+    except Exception:
+        config = get_multimodels_configs_from_hub(model_id)  # Applied on SD, encoder-decoder models
+    target_entry = ModelCacheEntry(model_id, config)
     # Extract model type: it will be used as primary key for lookup
     model_type = target_entry.config["model_type"]
     registry_folder = get_registry_folder_for_mode(mode)
     registry_pattern = registry_folder + "/" + model_type
     model_files = [path for path in repo_files if registry_pattern in path]
+    white_list = CACHE_WHITE_LIST  # All parameters except those in the whitelist must match
     model_entries = []
     with TemporaryDirectory() as tmpdir:
         for model_path in model_files:
             local_path = api.hf_hub_download(cache_repo_id, model_path, local_dir=tmpdir)
             with open(local_path) as f:
                 entry_config = json.load(f)
                 # Remove neuron config for comparison as the target does not have it
-                neuron_config = entry_config.pop("neuron")
-                # All parameters except those in the whitelist must match
-                white_list = ["_name_or_path", "transformers_version", "eos_token_id", "bos_token_id", "pad_token_id"]
+                model_entries = lookup_matched_entries(
+                    entry_config, target_entry, white_list, model_entries, model_type
+                )
+
+    return model_entries
+
+
+def lookup_matched_entries(entry_config, target_entry, white_list, model_entries, model_type):
+    is_matched = True
+    entry_config, target_entry_config, neuron_config = _prepare_config_for_matching(
+        entry_config, target_entry, model_type
+    )
+    for name, value in entry_config.items():
+        if isinstance(value, Dict):
+            for param in white_list:
+                value.pop(param, None)
+                target_entry_config[name].pop(param, None)
+            for term in set(entry_config[name]).intersection(set(target_entry_config[name])):
+                if entry_config[name][term] != target_entry_config[name][term]:
+                    is_matched = False
+        else:
+            if value != target_entry_config[name]:
+                is_matched = False
+    if is_matched:
+        neuron_config.pop("model_type", None)
+        model_entries.append(neuron_config)
+
+    return model_entries
+
+
+def _prepare_config_for_matching(entry_config, target_entry, model_type):
```

Comment:
nit: I would define this function above the other since it's being used by the other (for visibility)

Reply:
Will do!